### PR TITLE
fix: [STO-5739]: Ticket Settings Save Workflow

### PR DESCRIPTION
**Pull request by ahokinson**
#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
</details>

### CODE REVIEW

File: src/modules/85-sto/components/ExternalTickets/Settings/ExternalTicketSettings.tsx [hunk 1]
```diff
@@ -138,24 +150,32 @@ const ExternalTicketSettings: React.FC = () => {
...
+
+      if (newSettings.projectKey || newSettings.issueType) {
+        saveSettings.body.additional = {}
+        if (newSettings.projectKey) {
+          saveSettings.body.additional.projectKey = newSettings.projectKey
+        }
+        if (newSettings.issueType) {
+          saveSettings.body.additional.issueType = newSettings.issueType
+        }
+      }
```
Comment (chris-harness) [inline src/modules/85-sto/components/ExternalTickets/Settings/ExternalTicketSettings.tsx:177]: It's OK to save an empty object for additional. That'd make this logic a lot simpler.

File: src/modules/85-sto/components/ExternalTickets/Settings/ExternalTicketSettings.tsx [hunk 2]
```diff
@@ -138,24 +150,32 @@ const ExternalTicketSettings: React.FC = () => {
...
-      )
+
+      const saveSettings = {
+        queryParams: { accountId, projectId: projectIdentifier, orgId: orgIdentifier },
+        body: {
+          connectorId,
+          module,
+          service: 'Jira'
+        }
+      } as SettingsSaveSettingVariables
```
[the remark this answers is not in the document]
Comment (ahokinson) [reply src/modules/85-sto/components/ExternalTickets/Settings/ExternalTicketSettings.tsx:167]: I'm a little confused... You can't have conditional inclusion of a key without the structure I have. And while I understand having an empty additional object may be okay, it doesn't save us from conditionally including projectKey or issueType. They can't be undefined and an empty string shouldn't be used either.

[the remark this answers is not in the document]
Comment (ahokinson) [reply src/modules/85-sto/components/ExternalTickets/Settings/ExternalTicketSettings.tsx:167]: I assume you want me to do this inline, but I'm not familiar with a way I can do that.

[the remark this answers is not in the document]
Comment (chris-harness) [reply src/modules/85-sto/components/ExternalTickets/Settings/ExternalTicketSettings.tsx:167]: You can conditionally set a field: if the value is `undefined` it is equivalent to not specifying the field at all.

**Review by chris-harness — APPROVED** [review]

**Review by atmollo-harness — APPROVED** [review]

**Review by jsentongo — APPROVED** [review]

**Pull request merged**